ADR/0039-creating-pending-pipelineruns.md

```diff
@@ -0,0 +1,66 @@
+
+If we implement MPC limiting, it will make sure pipeline runs are not timeouting because of MPC pool being used by builds started earlier.
+
+### Example
+
+This is an example of controller that +- implements this MPC limiting log:
+
+```
+2024-11-20 14:04:44,646 prs_starter Thread-3 (my_capacity_counter) INFO Loaded MPC capacity: {'linux/arm64': 7, 'linux/ppc64le': 32, 'linux/s390x': 0}
+2024-11-20 14:04:49,659 prs_starter.my_reconciler Thread-2 (my_reconciler) DEBUG Reconciling my-comp-2-on-push-zz6sw/jhutar-tenant (429faf3a-620d-4ccd-ac89-c7ba4db11c90)
+2024-11-20 14:04:49,659 prs_starter.my_reconciler Thread-2 (my_reconciler) DEBUG Missing capacity for my-comp-2-on-push-zz6sw/jhutar-tenant (429faf3a-620d-4ccd-ac89-c7ba4db11c90) on linux/s390x
+2024-11-20 14:04:49,659 prs_starter.my_reconciler Thread-2 (my_reconciler) DEBUG Reconciling my-comp-3-on-push-hj5bw/jhutar-tenant (ba328d32-e912-4a2c-98e8-506937577912)
+2024-11-20 14:04:49,659 prs_starter.my_reconciler Thread-2 (my_reconciler) DEBUG Missing capacity for my-comp-3-on-push-hj5bw/jhutar-tenant (ba328d32-e912-4a2c-98e8-506937577912) on linux/s390x
+2024-11-20 14:04:57,309 prs_starter Thread-3 (my_capacity_counter) INFO Loaded MPC capacity: {'linux/arm64': 7, 'linux/ppc64le': 32, 'linux/s390x': 1}
+2024-11-20 14:04:59,659 prs_starter.my_reconciler Thread-2 (my_reconciler) DEBUG Reconciling my-comp-2-on-push-zz6sw/jhutar-tenant (429faf3a-620d-4ccd-ac89-c7ba4db11c90)
+2024-11-20 14:04:59,660 prs_starter.my_reconciler Thread-2 (my_reconciler) INFO Capacity available for my-comp-2-on-push-zz6sw/jhutar-tenant (429faf3a-620d-4ccd-ac89-c7ba4db11c90) on all considered platforms
+2024-11-20 14:04:59,660 prs_starter.my_reconciler Thread-2 (my_reconciler) INFO Starting my-comp-2-on-push-zz6sw/jhutar-tenant (429faf3a-620d-4ccd-ac89-c7ba4db11c90)
+2024-11-20 14:05:00,529 prs_starter.my_reconciler Thread-2 (my_reconciler) INFO The my-comp-2-on-push-zz6sw/jhutar-tenant (429faf3a-620d-4ccd-ac89-c7ba4db11c90) was started, not watching it anymore
+2024-11-20 14:05:00,529 prs_starter.my_reconciler Thread-2 (my_reconciler) DEBUG Reconciling my-comp-3-on-push-hj5bw/jhutar-tenant (ba328d32-e912-4a2c-98e8-506937577912)
+2024-11-20 14:05:00,529 prs_starter.my_reconciler Thread-2 (my_reconciler) DEBUG Missing capacity for my-comp-3-on-push-hj5bw/jhutar-tenant (ba328d32-e912-4a2c-98e8-506937577912) on linux/s390x
+```
+
+This is what was happening:
+
+* at 2024-11-20 14:04:44,646, we checked MPC capacity and s390x was fully used, so reconciling pipeline runs (that are in pending state) `my-comp-2-on-push-...` and `my-comp-3-on-push-...` did not started them
+* at 2024-11-20 14:04:57,309, we noticed MPC freed the capacity for s390x
+* at 2024-11-20 14:04:59,660 pipeline run `my-comp-2-on-push-...` was started, no luck for `my-comp-3-on-push-...` though
+
+Currently this is POC code only and besides lots of other things, we know about these issues that would need to happen:
```

Comment:
You refer to code but don't reference it. Code isn't necessary for an ADR as we are instead trying to understand the motivation behind a problem and decide on the defining properties of functionality.

Reply:
This is the code I have so far: https://gitlab.cee.redhat.com/jhutar/mystoneinst/-/blob/main/prs-controller.py?ref_type=heads

I just wanted to note that there are some blockers and I can also mention some problems I hit when creating and testing the code where I only have workarounds for them, not solutions.

Or I can just remove this section?